### DISCUSSION TITLE
How to add requirement from git?

### DISCUSSION THREAD

**Discussion by pkolebski** [bentoml/BentoML#1446] [Q&A] [answered]
Using ```infer_pip_packages=True``` I can pack service but I am getting error while building docker 
```
ERROR: Could not find a version that satisfies the requirement pytorch-ssd==0.1
ERROR: No matching distribution found for pytorch-ssd==0.1
The command '/bin/sh -c if [ -f /bento/bentoml-init.sh ]; then bash -c /bento/bentoml-init.sh; fi' returned a non-zero code: 1
```
When I explicit say which package to install as below:
```
pip_required_packages = [
    'torch==1.2.0',
    'torchvision==0.4.0',
    'Pillow==8.1.0',
    'numpy==1.19.5',
    'git+https://github.com/xBerryOS/pytorch-ssd',
    'imagehash==4.0'
]


@env(pip_dependencies=pip_required_packages)
@artifacts([PytorchModelArtifact('object_detection_model')])
class ObjectDetector(BentoService):
```
I am getting error during packing this service:
```
pkg_resources.extern.packaging.requirements.InvalidRequirement: Parse error at "'+https:/'": Expected stringEnd
```
What is a proper way to include this package?

**Comment by parano** [comment, marked as answer]
@pkolebski here are related documentation for how to use a pip package from git: https://docs.bentoml.org/en/latest/concepts.html#pypi-packages

Note that this is not supported in the current version 0.11.0, but will be available in the next release. In the next release, you will be able to use git repo with the standard PyPI package descriptor format:

```python
@bentoml.env(
  pip_packages=[
    'scikit-learn',
    'pandas @https://github.com/pypa/pip/archive/1.3.1.zip',
  ]
)
```

Or use your own `requirement.txt` file, e.g.: `@bentoml.env(requirements_txt_file='./requirements.txt')`
